lib/create-app.js

```diff
@@ -206,14 +206,17 @@ function initWebId (argv, app, ldp) {
   // without permission by including the credentials set by the Solid server.
   app.use((req, res, next) => {
     const origin = req.get('origin')
-    const trustedOrigins = argv.trustedOrigins
+    const trustedOrigins = ldp.getTrustedOrigins(req)
     const userId = req.session.userId
     // Exception: allow logout requests from all third-party apps
     // such that OIDC client can log out via cookie auth
     // TODO: remove this exception when OIDC clients
     // use Bearer token to authenticate instead of cookie
     // (https://github.com/solid/node-solid-server/pull/835#issuecomment-426429003)
-    if (!argv.host.allowsSessionFor(userId, origin, trustedOrigins) && !isLogoutRequest(req)) {
+    // We only want to do this check if strictOrigin is set to false, since we trust WebACL to handle origins otherwise
+    // If we don't allow authentication with cookies we can't handle ACLs properly wrt trusted origins
```

Comment:
That's not the right explanation. We can handle ACLs properly; with non-cookie auth mechanisms.

The full explanation is:

Authentication cookies are an optimization:
instead of going through the process of
fully validating authentication on every request,
we go through this process once,
and store its successful result in a cookie
that will be reused upon the next request.
However, that cookie can then be sent by any server,
even servers that have not gone through the proper authentication mechanism.
However, if trusted origins are enabled,
then any origin is allowed to take the shortcut route,
since malicious origins will be banned at the ACL checking phase.

Reply:
I'm having problem wrapping my head around the reasoning for this part of the code, so thank you for the explanation. I think it's a good explanation, so I added it to the PR, hope that's ok.